### PR TITLE
SIDM-6455 Redirect SSO users to their IdP login page

### PR DESCRIPTION
**Pull request by nikola-naydenov-hmcts**
[SIDM-6455](https://tools.hmcts.net/jira/browse/SIDM-6455)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -31,6 +31,7 @@ allprojects {
   sourceCompatibility = 11
   targetCompatibility = 11
 
+  ext['idam.api-spec.version'] = '3.6.0-RC'
```
Comment (jburke-idam) [inline build.gradle:34]: placeholder?

Comment (nikola-naydenov-hmcts) [reply build.gradle:34]: yeah

Comment (nikola-naydenov-hmcts) [reply build.gradle:34]: need the other pr to be merged

File: src/main/java/uk/gov/hmcts/reform/idam/web/strategic/SPIService.java [hunk 2]
```diff
@@ -170,7 +170,9 @@ protected ApiAuthResult authenticate(final String username, final String passwor
         } catch (HttpClientErrorException | HttpServerErrorException he) {
             httpStatus = he.getStatusCode();
             if (httpStatus == HttpStatus.FORBIDDEN || httpStatus == HttpStatus.UNAUTHORIZED || httpStatus == HttpStatus.NOT_FOUND) {
-                resultBuilder.errorCode(getErrorCode(he.getResponseBodyAsString()));
+                ErrorResponse errorResponse = getErrorResponse(he.getResponseBodyAsString());
+                resultBuilder.errorCode(errorResponse.getCode());
+                resultBuilder.errorInfo(errorResponse.getInfo());
```
Comment (jburke-idam) [inline src/main/java/uk/gov/hmcts/reform/idam/web/strategic/SPIService.java:175]: Note sure I get this change - where does the info come from?

**Pull request merged**